### PR TITLE
Throw TrinoException PERMISSION_DENIED in case of access denied

### PR DESCRIPTION
**Pull request by pajaks**
## Description

Change more meaningful exception in case of permission denied

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by pajaks** [comment]
ci hit https://github.com/trinodb/trino/issues/18134

File: lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java [hunk 1]
```diff
@@ -1911,6 +1914,9 @@ private void flushBuffer(boolean finished)
                 }
                 catch (AmazonServiceException e) {
                     failed = true;
+                    if (e.getErrorCode().equals(S3_ACCESS_DENIED_ERROR_CODE)) {
```
Comment (findinpath) [inline lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java:1917]: pls add the exception stacktrace to the description of the issue

File: lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java [hunk 2]
```diff
@@ -251,6 +253,7 @@ public class TrinoS3FileSystem
     private static final Set<String> GLACIER_STORAGE_CLASSES = ImmutableSet.of(Glacier.toString(), DeepArchive.toString());
     private static final MediaType DIRECTORY_MEDIA_TYPE = MediaType.create("application", "x-directory");
     private static final String S3_DEFAULT_ROLE_SESSION_NAME = "trino-session";
+    private static final String S3_ACCESS_DENIED_ERROR_CODE = "AccessDenied";
```
Comment (ebyhr) [inline lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java:256]: nit: I would move it to before/after `NO_SUCH_*_ERROR_CODE` constants and rename to `ACCESS_DENIED_ERROR_CODE`.

**Comment by ebyhr** [comment]
/test-with-secrets sha=3812b604b3a120f2d45af9dcb975be5973a3791e

**Review by electrum — CHANGES_REQUESTED** [review]
`PERMISSION_DENIED` is a user-facing error code that means the user didn't have permission to do something at the SQL layer. If the query passed authorization checks, then it's expected that accessing storage will succeed. An S3 access denied error is typically something for the administrator to resolve.

**Comment by electrum** [comment]
Also, `TrinoS3FileSystem` and the other Hadoop implementations for cloud storage are deprecated and in maintenance mode -- we shouldn't be adding features to them.

**Pull request closed without merging**